### ISSUE TITLE
Search by "General Contour"

### ISSUE THREAD

**Issue by martha-thomae** [ECHOES-from-the-Past/mei-analyser#95]
For "specific contour" search you have query values like `+1`, `-1`, `0`, `+2`, ... (i.e., integers).

Now, for the "general contour" search, your query values would be the following four characters: `u` (up), `d` (down), `s` (same), and `n` (neutral/unknown).

Let's define the melodic interval (`intm`) as the difference between a note's position and the previous note. In other words, `intm = loc_next - loc_prev`. Then you have three possible scenarios:
- When the result of `intm > 0`, then `u`
- When the result of `intm < 0`, then `d`
- When the result of `intm = 0`, then `s`

**Note:** In Aquitanian script (and square script), we will never have `n` as all the locations are known and that value is reserved for unknown heights.

----
Basically, you already have contour search implemented for Aquitanian and square. So now, for this general contour search, all positive intervals are `u`, all negative intervals are `d`, and all zero intervals are `s`.

**Comment by martha-thomae** [comment]
> **Note:** In Aquitanian script (and square script), we will never have n as all the locations are known and that value is reserved for unknown heights.

For the chants with unknown heights, we don't use `@pname` or `@loc` in MEI. Instead, we use `@intm` (see [att.intervalMelodic class](https://music-encoding.org/guidelines/v5/attribute-classes/att.intervalMelodic.html)). `@intm` encodes the melodic interval from the previous pitch using the values `u`, `d`, `s`, `n`, `sd`, and `ud` (see [INTERVAL.MELODIC data type](https://music-encoding.org/guidelines/v5/data-types/data.INTERVAL.MELODIC.html)).

In those cases, for chants that have `@intm` to encode things rather than the specific heights like in square (with `@pname`) and Aquitanian (with `@loc`) scripts, then we would use also the `n` value in the query. Fow now, we need an example of these chants to be able to implement this other thing. @elsinhadl, could you provide us with an example of a Hispanic script chant with at least three neumes? Thank you!

**Comment by notkaramel** [comment]
@martha-thomae I've implemented general contour since this is a relatively easy to implement feature :) Here's the new tooltip for contour: 

![image](https://github.com/user-attachments/assets/516e7988-9dd5-4efa-a177-e5a370ea9374)

I've added this in commit 0eb292c173dca02fcd028b0bc8eb3df5e3f48223 and going to merge directly to the current version of the interface for demo on our meeting!

**Comment by notkaramel** [comment]
@martha-thomae let's reopen this issue once we work with `@intm = n` or similar things later! :)

**Comment by martha-thomae** [comment]
Ways to indicate the "height" of a note: 
- Square notation: `<nc pname="c" oct="3">`
- Aquitanian: `<nc loc="0">`
- Old Iberian notation: `<nc intm="...">` (melodic interval attribute)

Example of Old Iberian notation neume:
```xml
<neume>
   <nc/>
   <nc intm="u"/>
   <nc intm="d"/>
   <nc intm="s"/>
   <nc intm="n"/>
</neume>
```
Documentation on `@intm` (melodic interval): https://music-encoding.org/guidelines/v5/data-types/data.INTERVAL.MELODIC.html

**Assumption:** The first `<nc>` in a `<neume>` will always be `@intm = n` (implicitly).

**Example query:** `n u d s n`
